File: math/digits/digit.cpp
#include <cmath>

//Extract a specific digit from a number
int ExtractDigits(int num, int digit) {
	int digit_column = pow(10, digit);
	return (num / digit_column) % 10;
}

// Simple Math
int CountDigits(int num) {
	
	int digits = 0;
	while (num > 0) {
		num /= 10;
		digits++;
	}
	return digits;
}

// Logarithms
int CountDigitsWithLog(int num) {	
	return floor(log10(n) + 1);
}

// Strings
int CountDigitsWithString(int num)
{
    string s = to_string(num);
    return s.size(); 
}

File: math/digits/digit.py
import math

# math
def count_digits(num):
	digits = 0
	while num > 0:
		num //= 10
		digits += 1
	
	return digits

# logarithms
def count_digits_log(num):
	return math.floor(math.log10(num) + 1)


# strings
def count_digits_string(num):
    return len(str(num))
